Remove all Silicon deprecations

waynew opened the pull request:
### What does this PR do?

Removes the bits that  were deprecated for Silicon

### What issues does this PR fix or reference?

#60893 

### Previous Behavior

Warning messages would appear about deprecated features

### New Behavior

Deprecated features are gone and you get errors instead.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

waynew committed:
Remove all deprecations

waynew commented:
@cbosdo I *think* this is the correct way to adjust the tests for this deprecated removal - if there's a better way, please let me know!

cbosdo reviewed (dismissed):
LGTM

s0undt3ch commented on changelog/60893.removed:
```diff
@@ -0,0 +1 @@
+Removed deprecated virt.migrate_non_shared ,virt.migrate_non_shared_inc, ssh from virt.migrate, and python2/python3 args from salt.utils.thin.gen_min and .gen_thin
```
```suggestion
Removed deprecated virt.migrate_non_shared, virt.migrate_non_shared_inc, ssh from virt.migrate, and python2/python3 args from salt.utils.thin.gen_min and .gen_thin
```

waynew replied:
Nice catch! :+1:

waynew committed:
Remove the leftover uses

There were a few places that didn't show up on previous searches. Also
fix the changelong.

cbosdo commented:
@rst0git could you have a look at this PR? you know that code better than anyone else :grin:

rst0git commented on salt/modules/virt.py:
```diff
@@ -5477,145 +5477,12 @@ def define_vol_xml_path(path, pool=None, **kwargs):
     :param target: target libvirt URI or host name
-    :param ssh: True to connect over ssh
 
        .. deprecated:: 3002
```
Perhaps we could drop this `.. deprecated:: 3002` comment as well?
